### ISSUE TITLE
Reranker minicpb-pairwise finetuning loss issue

### ISSUE THREAD

**Issue by zeruiz99** [FlagOpen/FlagEmbedding#644]
![image](https://github.com/FlagOpen/FlagEmbedding/assets/114459978/bc3916c9-ee4d-45ff-bb5a-81e4185852f3)

torchrun --nproc_per_node 1 \
-m FlagEmbedding.llm_reranker.finetune_for_layerwise.run \
--output_dir {path to save model} \
--model_name_or_path openbmb/MiniCPM-2B-dpo-bf16 \
--train_data ./toy_finetune_data.jsonl \
--learning_rate 2e-4 \
--num_train_epochs 1 \
--per_device_train_batch_size 2 \
--dataloader_drop_last True \
--query_max_len 512 \
--passage_max_len 512 \
--train_group_size 8 \
--logging_steps 1 \
--save_steps 2000 \
--ddp_find_unused_parameters False \
--gradient_checkpointing \
--deepspeed stage1.json \
--warmup_ratio 0.1 \
--bf16 \
--use_lora True \
--lora_rank 32 \
--lora_alpha 64 \
--use_flash_attn True \
--target_modules q_proj k_proj v_proj o_proj \
--start_layer 18 \
--head_multi True \
--head_type simple \
--lora_extra_parameters linear_head

the train data is a little bit too small like 50k lines, does this model fit better for larger train data or it will not converge

**Comment by yjoonjang** [comment]
Hi, @zeruiz99 
I'm dealing with the same problem. Can you tell me how you resolved it ?
